sars2seq/genome.py

```diff
@@ -211,16 +213,25 @@ def aaSequences(self, featureName):
         feature = self.features[featureName]
         name = feature['name']
 
-        referenceAa = AARead(
-            self.features.reference.id + f' ({name})',
-            feature.get('translation', translate(feature['sequence'], name)))
+        if (name == 'surface glycoprotein' and genomeNt.count('-') > 0 and
+                genomeNt.count('-') % 3 == 0):
```

Comment:
I don't know why you would count this twice. You program as if computing time and resources were all free and infinite. This is not what you would do in real life if I asked you are there more than zero hyphens in a 30K genome and you had to count them by hand and you knew in advance that you were immediately going to need the count again.

```python
gapCount = genomeNt.count('-')
if name == 'surface glycoprotein' and gapCount and gapCount % 3 == 0:
```

Reply:
Done.